### ISSUE TITLE
Some forms cannot be opened and edited in Visual Studio WinForms designer

### ISSUE THREAD

**Issue by lextm** [OpenLiveWriter/OpenLiveWriter#403]
Forms such as `SatelliteApplicationForm` have their internal logic that are not designer friendly and can trigger exceptions when opened in designer. Related to #394. All forms derived from them cannot be opened in designer either.

The code should be modified to support designer mode.

**Comment by kathweaver** [comment]
Do you know how to fix it?

**Comment by kathweaver** [comment]
I'm actually seeing this when I open SatelliteApplicationForm:

The designer could not be shown for this file because none of the classes within it can be designed. The designer inspected the following classes in the file: SatelliteApplicationForm --- The base class 'OpenLiveWriter.ApplicationFramework.ApplicationForm' cannot be designed.

**Comment by kathweaver** [comment]
I can open ApplicationForm

**Comment by lextm** [comment]
@kathweaver yes, by using design mode detection can solve it, https://msdn.microsoft.com/en-us/library/system.componentmodel.component.designmode.aspx. I will try to send a pull request when I have time, but anyone can step in before me.

**Comment by kathweaver** [comment]
Well, I haven't figured out to do it -- can you give more pointers?

**Comment by lextm** [comment]
@kathweaver Many old books on Windows Forms programming talked about that, https://books.google.com.hk/books?id=CpgQthE6v_0C&pg=PA247&lpg=PA247&dq=design+mode+windows+forms&source=bl&ots=VIRl0LxySr&sig=9agY-VT1z2Vlq4fMTqT5q2uNCJk&hl=en&sa=X&ved=0ahUKEwiTvsiftZzLAhVFrJQKHRxAA9o4ChDoAQggMAE#v=onepage&q=design%20mode%20windows%20forms&f=false

**Comment by kathweaver** [comment]
This is taking me a while to wrap my head around, so if anyone can fix this, please do, issue #394 depends on it.

**Comment by lextm** [comment]
@kathweaver I just created #406 to address that. Signing the CLA now.

After merging that you should be able to edit the form in designer.

**Comment by kathweaver** [comment]
Awesome, and I can read your code and learn stuff!

Thank you.

**Comment by lextm** [comment]
#406 fixes the project files. But still some forms fail to be opened in designer, as they don't have a default constructor defined (which the designer relies on). That requires some investigation to fix them all. 

@kathweaver for `PostEditorForm`, you can simply add the following constructor, and then the designer should work.
```
        public PostEditorForm()
        {
            CommonInit();
        }
```

Will see if I can send another pull request to address all such forms, but at least you get a chance to know the cause.

**Comment by kathweaver** [comment]
That didn't fix the problem.

**Comment by shanselman** [comment]
Thanks for reporting this and for helping Open Live Writer over the years. We are doing an issue-bankruptcy cleanup so we can make the repository maintainable again and focus on current, actionable work. This older issue is being closed as part of that cleanup, not because it was unimportant. If this still reproduces in the current code or is still relevant to the revitalization work, please open a fresh issue with current details and we will triage it going forward.
